forch/dva_state_collector.py

```diff
@@ -0,0 +1,94 @@
+"""Collect DVA related info"""
+
+import argparse
+import copy
+import json
+import logging
+import os
+import sys
+import yaml
+
+from forch.proto.dva_state_pb2 import DvaState, Switch, Port
+from forch.utils import configure_logging, proto_dict
+
+LOGGER = logging.getLogger('dstate')
+
+
+class DvaStateCollector:
+    """Process DVA related info"""
+    def __init__(self):
+        self._state = DvaState()
+
+    def process_behavioral_config(self, config):
+        """Process behavioral Faucet configuration"""
+        for switch_name, switch_info in config.get('dps', {}).items():
+            for port_id, port_info in switch_info.get('interfaces', {}).items():
+                native_vlan = port_info.get('native_vlan')
+                if not native_vlan:
+                    continue
+                acls = port_info.get('acls_in')
+
+                switch = self._state.switches.setdefault(switch_name, Switch())
+                port = switch.ports.setdefault(port_id, Port())
+
+                port.vlan = native_vlan
+                port.acls.extend(acls)
+
+    def process_authentication_state(self, switch_name, port_id, mac, auth_state,
+                                     ignore_non_existing_port=False):
+        """Process authentication state"""
+        port = self._state.switches.get(switch_name, {}).get(port_id)
+        if not ignore_non_existing_port and not port:
+            LOGGER.error(
+                'Cannot find switch and port definition in behavioral config: %s, %s',
+                switch_name, port_id)
+            return
+
+        if not port:
+            switch = self._state.switches.setdefault(switch_name, Switch())
+            port = switch.ports.setdefault(port_id, Port())
+
+        port.auth_states[mac] = auth_state
+
+    def get_dva_state(self):
+        """Get DVA state"""
+        return copy.deepcopy(self._state)
+
+
+def load_faucet_config(file):
+    """Load network state file"""
+    with open(file) as config_file:
+        return yaml.safe_load(config_file)
+
+
+def write_dva_state(dva_state, file):
+    """Write dva state to file"""
+    dva_state_json = proto_dict(dva_state)
+    with open(file, 'w') as dva_state_file:
+        json.dump(dva_state_json, dva_state_file)
+
+
+def parse_args(raw_args):
```

Comment:
I don't think this is a good approach -- not every module should have args parser -- it should only be for a few seslect entry points. This feature/modules isn't big enough to merit its own args.